java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogsTest.java

```diff
@@ -0,0 +1,357 @@
+package jmri.jmrit.display.layoutEditor;
+
+import java.awt.GraphicsEnvironment;
+import java.awt.geom.Rectangle2D;
+import jmri.util.JUnitUtil;
+import jmri.util.junit.rules.RetryRule;
+import org.junit.*;
+import org.junit.rules.Timeout;
+import org.netbeans.jemmy.operators.JButtonOperator;
+import org.netbeans.jemmy.operators.JDialogOperator;
+import org.netbeans.jemmy.operators.JFrameOperator;
+import org.netbeans.jemmy.operators.JTextFieldOperator;
+
+/**
+ * Test simple functioning of LayoutEditorDialogs
+ *
+ * @author	Paul Bender Copyright (C) 2016
+ */
+public class LayoutEditorDialogsTest {
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(10); // 10 second timeout for methods in this test class.
+
+    @Rule    // allow 2 retries of intermittent tests
+    public RetryRule retryRule = new RetryRule(2); // allow 2 retries
+
+    /*
+     * This is called once before all tests
+     */
+    @BeforeClass
+    public static void setUpClass() {
+        JUnitUtil.setUp();
+        if (!GraphicsEnvironment.isHeadless()) {
+            layoutEditor = new LayoutEditor();
+            layoutEditorDialogs = layoutEditor.getLEDialogs();
+            layoutEditor.setPanelBounds(new Rectangle2D.Double(0, 0, 640, 480));
+        }
+    }
+
+    /*
+     * This is called once after all tests
+     */
+    @AfterClass
+    public static void tearDownClass() {
+        if (!GraphicsEnvironment.isHeadless()) {
+            JUnitUtil.dispose(layoutEditor);
+            layoutEditor = null;
+            layoutEditorDialogs = null;
+        }
+        JUnitUtil.tearDown();
+    }
+
+    private static LayoutEditor layoutEditor = null;
+    private static LayoutEditorDialogs layoutEditorDialogs = null;
+
+    @Test
+    public void testCtor() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        Assert.assertNotNull("layoutEditor exists", layoutEditor);
+        Assert.assertNotNull("layoutEditorDialogs exists", layoutEditorDialogs);
+    }
+
+    @Test
+    public void testEnterGridSizes() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+
+        layoutEditorDialogs.enterGridSizes();
+        JFrameOperator jFrameOperator = new JFrameOperator(Bundle.getMessage("SetGridSizes"));
+        Assert.assertNotNull("jFrameOperator 1st exists", jFrameOperator);
+
+        new JButtonOperator(jFrameOperator, Bundle.getMessage("ButtonCancel")).doClick();  // NOI18N
+
+        layoutEditorDialogs.enterGridSizes();
+        jFrameOperator = new JFrameOperator(Bundle.getMessage("SetGridSizes"));
+        Assert.assertNotNull("jFrameOperator 2nd exists", jFrameOperator);
+
+        JTextFieldOperator primaryGridSizeTextFieldOperator = new JTextFieldOperator(jFrameOperator, 0);
```

Comment:
It is somewhat unreliable to ask for items from the frame by index, because if you add another one, or reorder the existing ones, the index can change.

It is better to link the label associated with the field to the label.
For example:
https://github.com/JMRI/JMRI/blob/0bf187b6b904f7b43626715756232a3dc9a56bc6/java/src/jmri/jmrit/consisttool/ConsistToolFrame.java#L91-L96

Reply:
What’s a Label? A JLabel? I don’t see any in the LayoutEditorDialogs sources… Should I add them so I can use them to link agains?

Reply:
This site [http://wiki.netbeans.org/Jemmy_FAQ#How_does_one_get.2Fknow_the_index_of_a_component.3F](url) suggests using `myTextField.setName` and `NameComponentChooser`; is that ok instead? (It’s one less line of code.)

Reply:
(I linked the labels as you suggested.)